dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -69,24 +69,44 @@
     // Tracks requests that are current executing in delegate and are not tracked in queuedCalls
     private final AtomicInteger queueSizeEstimate = new AtomicInteger(0);
     private final int maxQueueSize;
-    private final Counter queueSizeCounter;
+    private final Supplier<Counter> queueSizeCounter;
     private final Timer queuedTime;
     private final Supplier<ListenableFuture<Response>> limitedResultSupplier;
+    private volatile boolean queueUsed;
 
-    QueuedChannel(LimitedChannel delegate, String channelName, TaggedMetricRegistry metrics, int maxQueueSize) {
+    QueuedChannel(LimitedChannel delegate, String channelName, QueuedChannelInstrumentation metrics, int maxQueueSize) {
         this.delegate = new NeverThrowLimitedChannel(delegate);
         this.channelName = channelName;
         // Do _not_ call size on a ConcurrentLinkedDeque. Unlike other collections, size is an O(n) operation.
         this.queuedCalls = new ProtectedConcurrentLinkedDeque<>();
         this.maxQueueSize = maxQueueSize;
-        this.queueSizeCounter = DialogueClientMetrics.of(metrics).requestsQueued(channelName);
-        this.queuedTime = DialogueClientMetrics.of(metrics).requestQueuedTime(channelName);
+        // Lazily create the counter. Unlike meters, timers, and histograms, counters cannot be ignored when they have
+        // zero interactions because they support both increment and decrement operations.
```

Comment:
have opened https://github.com/palantir/tritium/pull/872 as a possible alternative to this... not sure whether the $ saving would be significant 🤷

Reply:
In favor of the general solution, I think QueuedChannel would still need to track state to guard `queuedTime` which is updated with `0` when the queue is bypassed, we avoid throwing zeros into it until it's first used.